### DISCUSSION TITLE
How to have a global sort based on Model full_name attribute?

### DISCUSSION THREAD

**Discussion by micraux** [filamentphp/filament#8046] [Help]
I have a attribute called `full_name` in my User model: it is just basically a `first_name` and `last_name` concatenation.

I would like my table to show the user `full_name` (which I can do) but also to be able to perform a global search on this attribute. But the search displays an error saying that the `full_name` column doesn't exist in the User table.

So I set up a little workaround: adding the `first_name` and `last_name` columns in my table, making them _searchable()_ and _toggleable()_ to hide them.

I am sure there is a more elegant way to achieve this...

**Comment by pboivin** [comment]
The searchable/toggleable is an interesting workaround! Alternatively, you could use a virtual column on your DB table instead of a model attribute. This will enable all table features like searching, sorting, filtering, etc.
https://kirschbaumdevelopment.com/insights/leveraging-virtual-generated-columns

**Comment by micraux** [comment]
Thank you Patrick!
I'll have a look to achieve that with a virtual column
